cmd/vulcan-zap/main.go

```diff
@@ -87,21 +88,52 @@ func main() {
 			users.SetUserEnabled("1", "0", "True")
 		}
 
-		log.Printf("Running spider %v levels deep...", opt.Depth)
+		logger.Debugf("Running spider %v levels deep...", opt.Depth)
 
 		client.Spider().SetOptionMaxDepth(opt.Depth)
 		resp, err := client.Spider().Scan(targetURL.String(), "", "", "", "")
 		if err != nil {
-			return err
+			return fmt.Errorf("error executing the spider: %v", err)
+		}
+
+		v, ok := resp["scan"]
+		if !ok {
+			// Scan has not been executed. Due to the ZAP proxy behaviour
+			// (the request to the ZAP API does not return the status codes)
+			// we can not be sure whether it was because a non existant target
+			// or because an error accessing the ZAP API. Therefore, we will
+			// terminate the check without errors.
+			logger.WithFields(logrus.Fields{"resp": resp}).Error("Scan not present in response body when calling Spider().Scan()")
```

Comment:
Maybe we discussed this and I forgot, but I would just log with "Warn" or even "Info" level here as we are not returning an error.

Reply:
Done